### PR TITLE
Do not commit to a branch before checking that there are differences.

### PR DESCRIPTION
**Pull request by fabianx-ai**

### CODE REVIEW

File: src/Drupal/Composer/DrupalInstallerPlugin.php [hunk 1]
```diff
@@ -520,6 +518,10 @@ protected function afterAllPatchesGitBranchCleanup(PackageInterface $package) {
         $this->verifyGitBranchExists($this->git['base-branch']);
 
         $isGitDiff = $this->isGitDiff($branchName);
+        if ($isGitDiff) {
+          $this->afterCommit($package);
+        }
```
Comment (fabianx-ai) [inline src/Drupal/Composer/DrupalInstallerPlugin.php:523]: This slipped in, but makes generally more sense to check if the branch is identical to the base branch.

File: src/Drupal/Composer/DrupalInstallerPlugin.php [hunk 2]
```diff
@@ -622,7 +624,7 @@ protected function getPackageVersion(PackageInterface $package) {
                 }
                 else {
                     // Drupal contrib versions have three numbers, i.e. 7.x-1.7.
-                    $version = $matches[1] . '.x-' . $matches[2] . '.' . $matches[3];
+                    $version = '7.x-' . $matches[1] . '.' . $matches[2];
```
Comment (douggreen) [inline src/Drupal/Composer/DrupalInstallerPlugin.php]: Will this ever be used on 8.x?

Comment (fabianx-ai) [reply src/Drupal/Composer/DrupalInstallerPlugin.php]: I am actually not sure, but I found no good way to see the actual channel used for the package. Also I guess 8.x packages might get semver at some point in the future, so maybe its less important there.

**Comment by bdragon28** [comment]
The only part of this PR that is still relevant now is the git branch check.

**Comment by douggreen** [comment]
Please review after my merge resolution. It looks like nothing is left.

**Comment by fabianx-ai** [comment]
I changed the title to reference just what the branch check does.

**Comment by fabianx-ai** [comment]
RTBM -> even though I wrote it this, is long enough ago that I can review this with a fresh perspective now and it looks still good to me.

**Pull request merged**